api/v1alpha3/azuremachine_types.go

```diff
@@ -72,6 +72,18 @@ type AzureMachineSpec struct {
 	// AllocatePublicIP allows the ability to create dynamic public ips for machines where this value is true.
 	// +optional
 	AllocatePublicIP bool `json:"allocatePublicIP,omitempty"`
+
+	// SpotVMOptions allows the ability to specify the Machine should use a Spot VM
+	// +optional
+	SpotVMOptions *SpotVMOptions `json:"spotVMOptions,omitempty"`
+}
+
+// SpotVMOptions defines the options relevant to running the Machine on Spot VMs
+type SpotVMOptions struct {
+	// MaxPrice defines the maximum price the user is willing to pay for Spot VM instances
+	// +optional
+	// +kubebuilder:validation:pattern="^[0-9]+(\.[0-9]+)?$"
+	MaxPrice *string `json:"maxPrice,omitempty"`
```

Comment:
Sorry I missed this earlier, why `MaxPrice` a string pointer? I think a float pointer would be more appropriate no? Maybe `*float64`

Reply:
Yeah it would be, I did this quite a while ago, I can't remember why I didn't do that, as that is what is described in the proposal too. Will investigate changing this

Reply:
I remember now why this can't be a float64. The CRD generator won't allow you to have a float64

https://github.com/kubernetes-sigs/controller-tools/blob/581f622859a03f041883454544414b1a179ebb8a/pkg/crd/schema.go#L409-L410

Reply:
I wonder if `// +kubebuilder:validation:Type=number` would be equivalent?